build.gradle

```diff
@@ -68,42 +69,44 @@ subprojects {
     }
 
     tasks.withType(JavaCompile).configureEach {
-      options.release.set(8)
+      if (isIntegrationTestOrSample) {
+        options.release.set(11)
+      } else {
+        options.release.set(8)
+      }
     }
   }
 
-  //noinspection UnnecessaryQualifiedReference
-  tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompilationTask).configureEach {
-    kotlinOptions {
-      if (parent?.path != ':integration-tests:mpp') {
-        // The configuration for KMP projects is different. We should fix that eventually.
-        //
-        // Caused by: groovy.lang.MissingPropertyException: Could not set unknown property 'jvmTarget' for object of type org.jetbrains.kotlin.gradle.tasks.internal.KotlinMultiplatformCommonOptionsCompat.
-        jvmTarget = JavaVersion.VERSION_1_8
+  tasks.withType(KotlinCompile).configureEach {
+    compilerOptions {
+      if (isIntegrationTestOrSample) {
+        jvmTarget.set(org.jetbrains.kotlin.gradle.dsl.JvmTarget.JVM_11)
+      } else {
+        jvmTarget.set(org.jetbrains.kotlin.gradle.dsl.JvmTarget.JVM_1_8)
       }
 
-      allWarningsAsErrors = rootProject.ext.makeWarningsErrors
-      freeCompilerArgs += ["-opt-in=kotlin.RequiresOptIn"]
+      allWarningsAsErrors.set(isCi && (useKotlinIr) && false)
```

Comment:
This comment/TODO seems like it should be preserved as well
```
// TODO: Remove when we remove the old backend.
  // TODO Repeatable: After cleaning up all deprecated methods remove the `&& false`.
```

I'm not sure if the first comment was just referring to `useKotlinIr` or not. @vRallev do you remember?